scratch/tif_sep.Rmd

```diff
@@ -0,0 +1,77 @@
+---
+title: "tif_sep"
+author: "Mikko Jimenez, Josh Carrell, and Caitlin Mothes"
+date: "'r Sys.Date()'"
+output: html_notebook
+---
+
+## Separate tif into different layers represented in one band
+
+## set up environment
+```{r}
+source("setup.R") 
+
+# read in raster to be separated 
+raster_path <- "data/input_raw/landcover_mchale/FC_landcover.tif"
+
+# write a lookup table that links landcover to the feature number in band
+lookup_table <- data.frame(
+  category_value = c(1, 2, 3, 4, 5, 6, 7),
+  land_cover = c("TreeCanopy", "Grass_Shrubs", "BareSoil", "Water", "Buildings", "Roads_Railroads", "OtherPaved")
+)
+
+# set output directory
+output_dir <- "data/input_raw/landcover_mchale/FC_landcover_sep/"
+```
+
+```{r}
+tif_sep <- function(raster_path, lookup_table, output_dir) {
+# read in McHale raster
+raster_data <- rast(raster_path)
+
+# create list to store shit
+layer_list <- list()
+
+# get unique values
+# remove NAs - sweet jesus this 
+unique_values <- unique(na.omit(values(raster_data)))
```

Comment:
A faster way to pull these values is: ``terra::freq(lc)$value`. (It was about 4 times faster when I benchmarked)